concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java

```diff
@@ -241,6 +241,26 @@ public static Throwable unwrapCompletionException(Throwable throwable)
     }
 
     /**
+     * Returns the result of the input {@link Future}, which must have already completed.
+     * <p>
+     * Similar to Futures{@link Futures#getDone(Future)}, but does not throw checked exceptions.
+     */
+    public static <T> T getDone(Future<T> future)
+    {
+        requireNonNull(future, "future is null");
+        checkArgument(future.isDone(), "future not done yet");
+        return getFutureValue(future);
+    }
+
+    /**
+     * Checks that the completed future completed successfully.
+     */
+    public static void checkSucceess(Future<?> future)
```

Comment:
I agree.  I'd change the signature to `checkSuccess(Future<?> future, String message)` and implement it like this:
```java
    public static void checkSuccess(Future<?> future, String errorMessageTemplate, Object... errorMessageArgs)
    {
        requireNonNull(future, "future is null");
        if (future.isDone()) {
            try {
                getFutureValue(future);
                return;
            }
            catch (RuntimeException ignored) {

            }
        }
        throw new IllegalArgumentException(String.format(errorMessageTemplate, errorMessageArgs));
    }
```
or maybe not error message formatting...  either way, but, I would only throw an illegal argument exception